src/control/SConscript

```diff
@@ -59,160 +28,109 @@ def is_release_build(benv):
 
 def get_build_flags(benv):
     """Return string of build flags"""
-    flags = ""
-    if not is_release_build(benv):
-        # enable race detector for non-release builds
-        flags += "-race"
-    else:
-        flags += '-buildmode=pie'
-    return flags
+    if is_release_build(benv):
+        return '-buildmode=pie'
+    # enable race detector for non-release builds
+    return '-race'
 
 
 def install_go_manpage(menv, name, binary):
     """
     Dynamically generate a man-page for a tool.
     """
-    gen_bin = join('$BUILD_DIR', 'src/control/bin', name)
-    name_ver = f'{name}.8'
-    build_path = join('$BUILD_DIR', 'src/control', name_ver)
-    install_path = join(menv.subst("$PREFIX"), 'share/man/man8', name_ver)
-    menv.Command(build_path, [gen_bin, binary], f'{gen_bin} manpage -o {build_path}')
-    menv.InstallAs(install_path, build_path)
+    gen_bin = join('$BUILD_DIR/src/control', name)
+    build_path = join('$BUILD_DIR/src/control', f'{name}.8')
+    menv.Command(build_path, [binary], f'{gen_bin} manpage -o {build_path}')
+    menv.Install('$PREFIX/share/man/man8', build_path)
+
+
+def gen_build_id():
+    """generate a unique build id per binary for use by RPM
+       https://fedoraproject.org/wiki/PackagingDrafts/Go#Build_ID"""
+    buildid = b2a_hex(urandom(20))
+    return '0x' + buildid.decode()
 
 
-# pylint: disable=too-many-arguments
-def install_go_bin(denv, gosrc, libs, name, install_name):
+def go_ldflags():
+    "Create the ldflags option for the Go build."
+
+    Import('daos_version', 'conf_dir')
+    path = 'github.com/daos-stack/daos/src/control/build'
+    return ' '.join([f'-X {path}.DaosVersion={daos_version}',
+                     f'-X {path}.ConfigDir={conf_dir}',
+                     f'-B $({gen_build_id()}$)'])
+
+
+def install_go_bin(denv, name, libs=None):
     """
     Build a Go binary whose source is under directory 'name' and install it
-    as 'install_name'.
     libs should be a list of scons-built libraries, or None if none are needed.
     """
-    # Repository path shared by DAOS Go package import paths
-    repopath = join("github.com", "daos-stack", "daos")
 
-    mod_src = join(gosrc, "cmd", name, "main.go")  # Module src
-    install_src = get_install_src_dir(repopath, name)
-    install_bin = get_install_bin_path(denv, install_name)
-    build_bin = join('$BUILD_DIR', 'src/control/bin', name)
-    src = [mod_src]
+    gosrc = Dir('.').srcnode().abspath
+    sources = Glob(f'cmd/{name}/*.go')
```

Comment:
Glob? I thought we discouraged this

Reply:
I think for c files we should be discouraging it, if you add a c file you should add it to the scons file or it won't get compiled.  Here we're adding dependencies so we need to use glob though as if you added a .go file and didn't update the scons then it wouldn't trigger rebuild even if the first build worked.